resolver.go

```diff
@@ -145,6 +145,14 @@ func (r *resolver) Start() error {
 		return fmt.Errorf("setting up IP table rules failed: %v", err)
 	}
 
+	// If the container goes down quickly after start resolver.Stop() can
+	// get called before Start() completes. Check if the socket is still
+	// valid before Activating the DNS server. This avoids a panic in dns
+	// libarary code because of missing nil check for interface's data.
+	// github.com/docker/docker/issues/28112
+	if r.conn == nil {
```

Comment:
`I am not sure I understand the opposition versus locks when we have the same structure being accessed/modified by different threads.`

For the reasons mentioned earlier its not needed for this issue. But to be safe we can lock around the resolver fields.